src/main/java/anodot/stage/destination/AnodotTarget.java

```diff
@@ -103,6 +110,12 @@ public void destroy() {
     public void write(Batch batch) throws StageException {
         try {
             Iterator<Record> records = batch.getRecords();
+            Record lastRecord = getLastRecord(batch);
+            if (lastRecord != null && !conf.agentOffsetUrl.equals("")) {
+                String pipelineId = lastRecord.get().getValueAsMap().get("tags").getValueAsMap().get("pipeline_id").getValueAsList().get(0).getValueAsString();
```

Comment:
Maybe there is a better way to get pipeline id? Even if we have to set additional parameters for that

Reply:
will be left as it is for now, as discussed